modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/mapping/smallcluster/AbstractTarget.java

```diff
@@ -34,7 +34,7 @@
  * <p>Children of this class are primary used for dispatching an execution to a corresponding
  * colocation method.
  */
-abstract class AbstractTarget implements ExecutionTarget {
+public abstract class AbstractTarget implements ExecutionTarget {
```

Comment:
it's not supposed to be public. If you don't like `finalise()` on ExecutionTarget, then it's better to do finalization inside of Factory whenever it required